b2/sync/folder.py

```diff
@@ -113,6 +114,20 @@ def ensure_non_empty(self):
         if not os.listdir(self.root):
             raise Exception('Directory %s is empty' % (self.root,))
 
+    def _check_file_access(self, local_path, reporter):
```

Comment:
`_is_file_accessible` would be a better name. You wouldn't need a docstring then.

As far as I know, the only safe way to safely handle file access checking, is to try to open it to actually do stuff and catch the exception when that happens. Let me show you an example:

Thread-1: _check_file_access -> True
Thread-2: change file permissions, or move the file, or remove the file, unmount the filesystem etc
Thread-1: attempt to access the file and fail ("unexpectedly")

Now, I think the main purpose of this method is to attempt to report read access to the reporter, as well as it is possible... But the comment or the docstring of it doesn't make it clear. Since you are refactoring it to an own method, could you please mention this? That way, when one day someone will want to use this method, he won't be misled into thinking that it actually solves the problem.